src/encoder.rs

```diff
@@ -1209,11 +1209,18 @@ pub fn encode_tx_block<T: Pixel, W: Writer>(
       residual,
       &ts.input_tile.planes[p].subregion(area),
       &rec.subregion(area),
-      visible_tx_w,
+      tx_size.width(),
       visible_tx_h,
     );
+    if visible_tx_w < tx_size.width() {
+      for row in residual.chunks_mut(tx_size.width()).take(visible_tx_h) {
+        for v in &mut row[visible_tx_w..] {
+          *v = 0;
+        }
+      }
+    }
   }
-  let visible_area = visible_tx_w * visible_tx_h;
+  let visible_area = tx_size.width() * visible_tx_h;
```

Comment:
Probably better to use different variable name than 'visilble_area', such as 'tx_area'?